### PR TITLE
chore: make private fields readonly

### PR DESCRIPTION
**Pull request by Tuditi**
# Description of change

Changes the types of a class from private to readonly, so the dev experience of TS users increases.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Tested within firefly.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/nodejs/CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Constructor types in `RegularTransactionEssence`;
 - `SenderFeature.getSender()` and `IssuerFeature.getIssuer()` now return the correct types;
+- Private properties on classes are now readonly.
```
Comment (thibault-martinez) [inline bindings/nodejs/CHANGELOG.md]: Either add an explanation what it fixes, or put it in `Changed` pls 🙏🏻

File: bindings/nodejs/lib/types/wallet/prepared-transaction.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export class PreparedTransaction {
     }
 
     /**
-     * This function signs a prepared transaction essence using the account's private key and returns
+     * This function signs a prepared transaction essence using the account's readonly key and returns
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/wallet/prepared-transaction.ts]: :trollface:

File: bindings/nodejs/CHANGELOG.md [hunk 3]
```diff
@@ -26,12 +26,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Constructor types in `RegularTransactionEssence`;
 - `SenderFeature.getSender()` and `IssuerFeature.getIssuer()` now return the correct types;
 
-
 ## 1.0.1 - 2023-07-25
 
 ### Changed
 
 - Deprecate `Account::{buildAliasOutput(), buildBasicOutput(), buildFoundryOutput(), buildNftOutput()}` and their `BuildAliasOutputData, BuildBasicOutputData, BuildFoundryOutputData, BuildNftOutputData` parameter types;
+- Private properties on classes are now readonly.
```
Comment (thibault-martinez) [inline bindings/nodejs/CHANGELOG.md]: Not in the right release

**Review by thibault-martinez — DISMISSED** [review]

File: bindings/nodejs/lib/types/block/input/input.ts [hunk 4]
```diff
@@ -35,7 +35,7 @@ class TreasuryInput extends Input {
     /**
      * The milestone id of the input.
      */
-    milestoneId: HexEncodedString;
```
Comment (abdulmth) [inline bindings/nodejs/lib/types/block/input/input.ts]: This is a breaking change, public to readonly, no?

**Review by abdulmth — CHANGES_REQUESTED** [review]

File: bindings/nodejs/lib/types/block/input/input.ts (hunk 4 above)
Comment (thibault-martinez) [reply bindings/nodejs/lib/types/block/input/input.ts]: Hmm good catch, there are actually more changes like that in the PR

Comment (abdulmth) [reply bindings/nodejs/lib/types/block/input/input.ts]: Yes, all of them should stay public

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/block/input/input.ts]: @abdulmth Can you open an issue where we'd list all breaking changes we'd do for a nodejs 2.0 version please? And add this

File: bindings/nodejs/lib/types/block/payload/transaction/unlock.ts [hunk 5]
```diff
@@ -37,7 +37,7 @@ class SignatureUnlock extends Unlock {
      * The signature.
      */
     @Type(() => Ed25519Signature)
-    signature: Ed25519Signature;
+    readonly signature: Ed25519Signature;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/block/payload/transaction/unlock.ts]: 👀

File: bindings/nodejs/lib/types/block/signature.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ enum SignatureType {
 }
 
 abstract class Signature {
-    private type: SignatureType;
+    type: SignatureType;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/block/signature.ts]: readonly?

Comment (abdulmth) [inline bindings/nodejs/lib/types/block/signature.ts]:
```suggestion
    readonly type: SignatureType;
```

**Review by abdulmth — CHANGES_REQUESTED** [review]

**Review by abdulmth — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Comment by thibault-martinez** [comment]
(ignore the latest comment, GH bug)

**Pull request merged**